### PR TITLE
update github.com/emicklei/go-restful and github.com/containerd/containerd

### PR DESCRIPTION
**Pull request by captncraig**
These updates should resolve a few CVE warnings found by certain tools.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -492,6 +493,9 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 )
 
+// replacements for security updates
+replace github.com/emicklei/go-restful => github.com/emicklei/go-restful/v3 v3.8.0
```
Comment (rfratto) [inline go.mod:497]: Replace directives are ideally temporary workarounds, and I'm hesitant to add more of them, especially for security updates. 

Is it necessary to use the v3 package? It looks like go-restful had a v2.16.0+incompatible release [15 days ago](https://github.com/emicklei/go-restful/releases/tag/v2.16.0).

Comment (rfratto) [reply go.mod:497]: I tried removing the replace directive and everything still builds. Was this needed for something in particular?

Comment (captncraig) [reply go.mod:497]: Trivy shows a critical vulnerability in `v2.9.5+incompatible`, resolved in `v3.8.0`. Again this looks like a case where the tool's definitions are not smart enough to know that you've fixed it.

Comment (rfratto) [reply go.mod:497]: Does it still show it in v2.16.0+incompatible?

Comment (captncraig) [reply go.mod:497]: yes it does :(
and I think it is unlikely to count on kube-openapi updating to v3 very easily:

```
go mod why -m github.com/emicklei/go-restful
go: downloading github.com/Microsoft/hcsshim v0.9.2
go: downloading github.com/miekg/pkcs11 v1.0.3
# github.com/emicklei/go-restful
github.com/grafana/agent/cmd/agentctl
sigs.k8s.io/controller-runtime/pkg/client/apiutil
k8s.io/client-go/discovery
k8s.io/client-go/openapi
k8s.io/kube-openapi/pkg/handler3
k8s.io/kube-openapi/pkg/common
github.com/emicklei/go-restful
```

Comment (rfratto) [reply go.mod:497]: `k8s.io/kube-openapi/pkg/common` uses v3 now: https://github.com/kubernetes/kube-openapi/commit/ca8f27ec61da5128d224678153c3fcd976d4afa5

Can we update that module?

Comment (captncraig) [reply go.mod]: It works with that! no replacements!

**Review by rfratto — APPROVED** [review]
Looks good, but can we note this as an other change in the changelog with a report of which CVEs this removes from reporting tools? 

(And also note that they weren't directly accessible from the agent so people don't worry)

**Pull request merged**